debarshi/js/components/todoBase.js

```diff
@@ -0,0 +1,55 @@
+import todos from "../constant.js"
+import render from "../todo.js"
+import { todoWrapper } from "../selectors.js"
+
+const createTodoBase = (todo, props = {}) => {
+  const { draggable = false } = props
+
+  const onTodoDragStart = (event) => {
+    const dataId = event.currentTarget.getAttribute("data-id")
+    event.dataTransfer.dropEffect = "move"
+    event.dataTransfer.setData("data_id", dataId)
+  }
+
+  const onTodoDragOver = (event) => {
+    event.preventDefault()
+  }
+
+  const onTodoDrop = (event) => {
+    const currentId = event.currentTarget.getAttribute("data-id")
+    const previousId = event.dataTransfer.getData("data_id")
+
+    const currentTodoIdx = todos.findIndex(
+      (todo) => todo.id === Number(currentId),
+    )
+    const previousTodoIdx = todos.findIndex(
+      (todo) => todo.id === Number(previousId),
+    )
+
+    if (currentTodoIdx >= 0 && previousTodoIdx >= 0) {
+      const temp = todos[currentTodoIdx]
```

Comment:
can use destructing assignment here for swapping to avoid temp - Eg: [b,a] = [a,b]